src/Symfony/Component/OptionsResolver/OptionsResolver.php

```diff
@@ -896,6 +887,45 @@ public function offsetGet($option)
     }
 
     /**
+     * @param string $type
+     * @param mixed  $value
+     * @param array  $invalidTypes
+     *
+     * @return bool
+     */
+    private function verifyTypes($type, $value, array &$invalidTypes = array())
+    {
+        if ('[]' === substr($type, -2) && is_array($value)) {
+            $originalType = $type;
+            $type = substr($type, 0, -2);
+            $invalidValues = array_filter( // Filter out valid values, keeping invalid values in the resulting array
+                $value,
+                function ($value) use ($type) {
+                    return (function_exists($isFunction = 'is_'.$type) && !$isFunction($value)) || !$value instanceof $type;
+                }
+            );
+
+            if (!$invalidValues) {
+                return true;
+            }
+
+            $invalidTypes[] = $this->formatTypeOf($value, $originalType);
+
+            return false;
+        }
+
+        if ((function_exists($isFunction = 'is_'.$type) && $isFunction($value)) || $value instanceof $type) {
+            return true;
+        }
+
+        if (!$invalidTypes) {
+            $invalidTypes[] = $this->formatTypeOf($value, null);
```

Comment:
What about using the types as keys instead and then get rid of the `array_unique()` keys (replace it with `array_keys()`)?

Reply:
fixed